pinot-common/src/main/java/org/apache/pinot/common/function/scalar/IpAddressFunctions.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.common.function.scalar;
+
+import com.google.common.net.InetAddresses;
+import java.math.BigInteger;
+import java.net.UnknownHostException;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+
+public class IpAddressFunctions {
+
+  private IpAddressFunctions() {
+  }
+
+  /**
+   * Returns true if ipAddress is in the subnet of ipPrefix
+   * ipPrefix is in cidr format (IPv4 or IPv6)
+   */
+  @ScalarFunction
+  public static boolean isSubnetOf(String ipPrefix, String ipAddress)
+      throws UnknownHostException {
+    if (!ipPrefix.contains("/")) {
+      throw new IllegalArgumentException("Invalid IP prefix: " + ipPrefix);
+    }
+    byte[] address;
+    byte[] argAddress;
+    int subnetSize;
+
+    String[] prefixLengthPair = ipPrefix.split("/");
+    try {
+      address = InetAddresses.forString(prefixLengthPair[0]).getAddress();
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("Invalid IP prefix: " + ipPrefix);
+    }
+    subnetSize = Integer.parseInt(prefixLengthPair[1]);
+    if (subnetSize < 0 || address.length * 8 < subnetSize) {
+      throw new IllegalArgumentException("Invalid IP prefix: " + ipPrefix);
```

Comment:
suggest printing `subnetSize` and `address.length` for easier debugging

Reply:
Done.